### PR TITLE
Intent To Implement: Directadvert support for amp-ads

### PR DESCRIPTION
**Pull request by dvdyakonov**
Added support for Directadvert widgets to amp-ad elements

### CODE REVIEW

File: ads/directadvert.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ */
+
+import {writeScript, validateData} from '../3p/3p';
+
+/**
+ * @param {!Window} global
+ * @param {!Object} data
+ */
+export function directadvert(global, data) {
+    validateData(data, ['blockId']);
```
Comment (zhouyx) [inline ads/directadvert.js]: nit: indentation 2

File: ads/directadvert.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+    const anchorEl = global.document.createElement('div');
+    anchorEl.id = `DIV_DA_${data['blockId']}`;
+    global.document.getElementById('c').appendChild(anchorEl);
+
+    writeScript(global, `https://code.directadvert.ru/data/${data['blockId']}.js?async=1&div=DIV_DA_${data['blockId']}`);
```
Comment (zhouyx) [inline ads/directadvert.js]: nit: 80 chars limit

Comment (zhouyx) [reply ads/directadvert.js]: Use `encodeURIComponent(data['blockId'])`.

File: ads/directadvert.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
+/**
+ * Copyright 2015 The AMP HTML Authors. All Rights Reserved.
```
Comment (zhouyx) [inline ads/directadvert.js]: nit: 2017

File: ads/directadvert.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
+<!---
+Copyright 2015 The AMP HTML Authors. All Rights Reserved.
```
Comment (zhouyx) [inline ads/directadvert.md]: nit: 2017

File: ads/directadvert.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    data-block-id="1953045">
+</amp-ad>
+
+```
+
+## Configuration
+
+For more information, please [see FAQ](https://www.directadvert.ru/text/help).
+
+Supported parameters:
```
Comment (zhouyx) [inline ads/directadvert.md]: Please indicate this is required parameter.

File: validator/testdata/feature_tests/ads.html [hunk 6]
```diff
@@ -116,6 +116,7 @@
         <option>csa</option>
         <option>custom</option>
         <option>eas</option>
+        <option>directadvert</option>
```
Comment (zhouyx) [inline validator/testdata/feature_tests/ads.html]: I don't think you need to change this file. It's used for validator test only. 

to @choumx I see you removed the validator test to our example/* files in #10292, do we ever update files in `validator/testdata/feature_tests` folder?

File: ads/_config.js [hunk 7]
```diff
@@ -267,6 +267,10 @@ export const adConfig = {
     renderStartImplemented: true,
   },
 
+  directadvert: {
+    renderStartImplemented: true,
```
Comment (zhouyx) [inline ads/_config.js]: I see you have renderStart implemented, but when I test locally there's no `render-start` message received. Could you please double check? Thanks

File: ads/directadvert.js (hunk 3 above)
Comment (dvdyakonov) [reply ads/directadvert.js]: Fixed

File: ads/directadvert.js (hunk 1 above)
Comment (dvdyakonov) [reply ads/directadvert.js]: Fixed

File: ads/directadvert.js (hunk 2 above)
Comment (dvdyakonov) [reply ads/directadvert.js]: Fixed

File: ads/directadvert.md (hunk 5 above)
Comment (dvdyakonov) [reply ads/directadvert.md]: Fixed

File: ads/_config.js (hunk 7 above)
Comment (dvdyakonov) [reply ads/_config.js]: I removed this option for directadvert

Comment (zhouyx) [reply ads/_config.js]: Note that we do recommend implementing renderStart API as it provides better UX. Please consider implementing it. 
Please refer to our [doc](https://github.com/ampproject/amphtml/blob/master/ads/README.md#available-apis) and let me know if there's any question.

**Comment by zhouyx** [comment]
@dvdyakonov I am seeing 32 files changes. Please rebase to master before we proceed with your PR.

**Comment by zhouyx** [comment]
LGTM. Thanks for the Pull Request.

**Pull request merged**